internal/tiltfile/k8s.go

```diff
@@ -281,7 +281,14 @@ func (s *tiltfileState) k8sResource(thread *starlark.Thread, fn *starlark.Builti
 	}
 
 	resourceName := workload.String()
-	manuallyGrouped := false
+	o, ok := s.k8sResourceOptions[resourceName]
+	if !ok {
+		// Set default options.
+		o.autoInit = true
+		o.tiltfilePosition = thread.CallFrame(1).Pos
```

Comment:
I think the only use we still have for this field is the error when k8s_resource specifies a `new_name` that's already taken: https://github.com/tilt-dev/tilt/blob/d06c6881a9a614c3814212e71c8d60dbfd9e91be/internal/tiltfile/tiltfile_state.go#L706

For this error message, we don't want the first k8s_resource for the resource, but the one that specifies `new_name`. What do you think of:
1. Delete other uses of this field
2. Rename this field to something like `newNameTiltfilePosition`
3. Set this on every call, if newName is set

(or maybe however we solve the previous comment makes this all irrelevant)
(or just remove the field entirely - it's kind of an annoying amount of bookkeeping for what's probably not much value)

Reply:
ya, i think this bookkeeping isn't work keeping. removed!